pkg/customresourcestate/registry_factory_test.go

```diff
@@ -173,6 +173,17 @@ func Test_values(t *testing.T) {
 			newEachValue(t, 2, "type", "type-a", "active", "1"),
 			newEachValue(t, 4, "type", "type-b", "active", "3"),
 		}},
+		{name: "path-relative valueFrom value", each: &compiledGauge{
```

Comment:
Can we add another Test  for this path

```
	if got == nil {
		if c.NilIsZero {
			return &eachValue{
				Labels: labels,
				Value:  0,
			}, nil
		}
```

Reply:
This is actually another bug in and of itself. Basically, the same one but for multi-length `valueFrom` paths. In the spirit of keeping the changesets digestible and throughly tested and reviewed in the CRS realm, I thinking if it'd be a better idea to target that in another PR?
***
JFYI the test below returns `nil` for both `main` and this PR.
```go
		{
			name: "multiple path-relative valueFrom values", each: &compiledGauge{
				compiledCommon: compiledCommon{
					path: mustCompilePath(t, "spec"),
					labelFromPath: map[string]valuePath{
						"version": mustCompilePath(t, "version"),
					},
				},
				ValueFrom: mustCompilePath(t, "order", "-1", "id"),
			}, wantResult: []eachValue{
				newEachValue(t, 3, "version", "v0.0.0"),
			},
		},
```

Reply:
Sounds good to me! Can you add a TODO to the code path?